prospector/commit_processor/feature_extractor.py

```diff
@@ -67,41 +61,20 @@ def extract_time_between_commit_and_advisory_record(
     return commit.timestamp - advisory_record.published_timestamp
 
 
-def extract_changes_relevant_path(
+def extract_changed_relevant_paths(
     commit: Commit, advisory_record: AdvisoryRecord
-) -> bool:
+) -> Set[str]:
     """
-    Decides whether any of the changed paths (by a commit) are in the list
+    Return the list of the changed paths (by a commit) which are in the list
     of relevant paths (mentioned in the advisory record)
     """
-    return any(
-        [changed_path in advisory_record.paths for changed_path in commit.changed_files]
-    )
+    return set(advisory_record.paths) & set(commit.changed_files)
```

Comment:
I think here the logic should be modified to match paths even when a path in the advisory record is a substring of a path modified in the commit.

For example, if the adivisory mentions SomeFile.java, and the commit at hand changes `src/java/whatever/SomeFile.java`, then this should be returned in the results. Not sure it is the case in the current implementation.

Reply:
I fixed it.